src/hazardous/base.rs

```diff
@@ -0,0 +1,156 @@
+use crate::errors::UnknownCryptoError;
+use std::{convert::TryFrom, fmt, marker::PhantomData};
+
+/// Marker trait for when a type contains some sensitive information.
+pub trait Secret {}
+
+/// Marker trait for when a type contains only non-sensitive information.
+/// Be careful if implementing this trait on your own. It cannot
+/// cause memory unsafety, and so is not marked `unsafe`. Implementing
+/// it can, however, lead to data types containing sensitive data ending
+/// up with APIs meant only for types containing only non-sensitive data.
+pub trait Public {}
+
+/// A small trait containing static information about the minimum and
+/// maximum size (in bytes) of a type containing data.
+pub trait Bounded {
+    /// The largest number of bytes this type should be allowed to hold.
+    const MIN: Option<usize> = None;
```

Comment:
I see no reason to leave these as Option. Aren't we always going to define a bound over all types?

Reply:
Yeah I can't think of a case when we'd want it. I actually don't remember why I put it there in the first place. At the time I was seven layers deep in thinking about the trait system, so this didn't get a ton of thought haha.